ui/library/src/utils/scroll.ts

```diff
@@ -106,12 +106,30 @@ export function getScrollParent(node: HTMLElement): Nullable<HTMLElement> {
   if (!node) {
     return document.documentElement;
   }
-  if (node.scrollHeight > node.clientHeight) {
+  if (isElementScrollable(node)) {
     return node;
   }
   return getScrollParent(node.parentElement as HTMLElement);
 }
 
+function isElementScrollable(node: HTMLElement): boolean {
+  if (!node) {
+    return false;
+  }
+
+  if (node.scrollTop > 0) {
+    return true;
+  }
+  node.scrollTop++;
+
+  //If the element cannot be scrolled, the scrolltop setting will not take effect and will be set to 0
+  const top = node.scrollTop;
+  //Reset scroll position
+  top && (node.scrollTop = 0);
```

Comment:
This not only causes a reflow, it'll cause the page to scroll to the top. In our case, that could cause anything listening for scroll to fire, and because of the directionality, it could potentially cause the header to hide.